### PR TITLE
js training

### PR DESCRIPTION
**Pull request by hondasmx**

### CODE REVIEW

File: res/learnProjects/javascript/LearnProject/.idea/workspace.xml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="BuildServerSettings">
+    <option name="LOGIN" value="" />
+    <option name="SERVER_URL" value="http://buildserver" />
```
Comment (karashevich) [inline res/learnProjects/javascript/LearnProject/.idea/workspace.xml]: Does this file need to be in LearnProject?

File: src/training/learn/lesson/javascript/ActionList.txt [hunk 2]
```diff
@@ -0,0 +1,516 @@
+ 0 = "ShowProjectStructureSettings"
```
Comment (karashevich) [inline src/training/learn/lesson/javascript/ActionList.txt]: Why do you need this list?

File: src/training/lang/JavaScriptLangSupport.kt [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        TODO("not implemented")
+    }
+
+    override fun applyToProjectAfterConfigure(): (Project) -> Unit = {
+    }
+
+    override fun checkSdk(sdk: Sdk?, project: Project) {}
+
+    override fun createProject(projectName: String, projectToClose: Project?): Project? {
+        return ProjectUtils.importOrOpenProject("/learnProjects/javascript/LearnProject", "LearnProject")
```
Comment (karashevich) [inline src/training/lang/JavaScriptLangSupport.kt]: Due to plugin design (unfortunately) `LearnProject` shouldn't be the same for the different languages excepting Java. Would we like to rename it like `JavaScriptLearnProject`? This problem appears only in a multilanguage layout like IntelliJ Ultimate with JS plugin.

File: src/training/learn/lesson/kimpl/LessonContext.kt [hunk 4]
```diff
@@ -4,6 +4,7 @@ import com.intellij.openapi.editor.Editor
 import com.intellij.openapi.project.Project
 import org.jetbrains.annotations.CalledInAwt
 import training.commands.kotlin.TaskContext
+import training.learn.lesson.LessonManager
```
Comment (karashevich) [inline src/training/learn/lesson/kimpl/LessonContext.kt]: Redundant import.

**Review by karashevich — COMMENTED** [review]
I didn't find enabling of optional dependency for language support. Maybe some commits are missing?
Something like `<depends optional="true" config-file="javascript-deps.xml">JavaScript</depends>` in a `plugin.xml` file.

**Comment by anstarovoyt** [comment]
@denofevil

**Pull request closed without merging**